✨ feat(chat): add AgentMessage component

kumanoayumi opened the pull request:
## Issue

- resolve:

## Why is this change needed?
<!-- Please explain briefly why this change is necessary -->
This PR introduces the AgentMessage component with proper syntax highlighting capabilities for code blocks in chat messages.

## What would you like reviewers to focus on?
<!-- What specific aspects are you requesting review for? -->

## Testing Verification
<!-- Please describe how you verified these changes in your local environment using text/images/video -->

storybook: https://liam-storybook-git-ui-chatagent-message-route-06-core.vercel.app/?path=/docs/components-chat-agentmessage--docs&globals=backgrounds.value:!hex(333333)

https://github.com/user-attachments/assets/ff526d85-b323-4481-81c0-aa03cf02d492

app: https://liam-app-git-ui-chatagent-message-route-06-core.vercel.app/app/projects/c4341447-3192-45cd-bec6-79d5377b1600/ref/main/build

https://github.com/user-attachments/assets/41d8a0b5-190f-47a4-9c36-f9dc470be234

## What was done
<!-- This section will be filled by PR-Agent when the Pull Request is opened -->

### 🤖 Generated by PR Agent at a163f04191b2c218af10e10f413e386913dbd95a

- Introduce `AgentMessage` chat component with agent-specific styling
  - Supports markdown and syntax-highlighted code blocks
  - Handles "generating" state with animated feedback
- Integrate `AgentMessage` into chatbot UI, replacing old bot message rendering
  - Unified markdown and code rendering for bot messages
  - Improved timestamp and agent avatar handling
- Add custom syntax highlighting theme and styles using CSS variables
  - New theme, CSS, and export in UI package
- Add Storybook stories for `AgentMessage` with rich markdown/code samples
- Update dependencies for security and feature support
  - Add `react-syntax-highlighter`, update `prismjs` to 1.30.0


## Detailed Changes
<!-- This section will be filled by PR-Agent when the Pull Request is opened -->

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>10 files</summary><table>
<tr>
  <td><strong>AgentMessage.tsx</strong><dd><code>New AgentMessage component with markdown and syntax highlighting</code></dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-e43d45645f7f77c3b8ea3a5eacde31f9fe98698f2f1bb801fd427aa6e256f3a7">+124/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>AgentMessage.module.css</strong><dd><code>Styles for AgentMessage, markdown, and code blocks</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-4be25f157abbb99c08b1c967fdb5d1e2c09081360ca69cceddee1bf7207f993e">+527/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>index.ts</strong><dd><code>Export AgentMessage and types for external use</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-fd905cd91ed66fc1a35eb0359aad5052b3002fd6606328dc242e7c5861aee571">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ChatMessage.tsx</strong><dd><code>Use AgentMessage for bot messages, add syntax highlighting</code></dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-51f002f7d0a0e88410ad3c8f59ba7953195055421846094c242737362273569e">+70/-9</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ChatbotDialog.tsx</strong><dd><code>Remove old loading indicator, use AgentMessage generating state</code></dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-40118adc6e918b1627ced03c99d854d215c13ec9f74f94e360cfc1f178373716">+5/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>syntax-theme.ts</strong><dd><code>Custom syntax highlighting theme using CSS variables</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-50a31dd972b47597f783989c965837b94564da055d20e508cf2c3160241f473d">+58/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>syntax-highlight.css</strong><dd><code>CSS variables for syntax highlighting colors and layout</code>&nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-d777b968852d61d1a7c90051edfc2ba06bbd8c4393c00756403b55e6d8903950">+25/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>index.ts</strong><dd><code>Export syntax-theme from UI styles</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-de280f261490fa65c8dff2b4f5fd6e71a9f3db36040de9a2f5cb4c70af06f240">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>globals.css</strong><dd><code>Import syntax highlighting CSS into global styles</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-0ae416cde0b2c1884bccd88880c4e4f0717e79e0105d61abcf6fd7aa1184b26c">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>index.ts</strong><dd><code>Export styles from UI package root</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-b29ec36cd11535c8a77bf7bd30ebc1c4b8f6855af8b2347cd18fc887dd081bca">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>AgentMessage.stories.tsx</strong><dd><code>Storybook stories for AgentMessage with markdown/code</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-636aaef145a1f38b1beb5b2f97408aadb4881f2c8765dd54d0c2625c2a31cd67">+492/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td><strong>package.json</strong><dd><code>Add react-syntax-highlighter and types, update prismjs</code>&nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-1cb683806b3ef955a38fae27b87a08febf12a8b1465dedad3a3f92c0136f132a">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>package.json</strong><dd><code>Override prismjs to 1.30.0 for security</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>pnpm-lock.yaml</strong><dd><code>Lockfile updates for new/updated dependencies</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/liam-hq/liam/pull/1653/files#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bb">+165/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

## Additional Notes
<!-- Any additional information for reviewers -->

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

MH4GF commented on frontend/apps/app/components/ChatbotButton/components/ChatMessage/ChatMessage.tsx:
```diff
@@ -31,6 +58,40 @@ export const ChatMessage: FC<ChatMessageProps> = ({
       })
     : null
 
+  // For bot messages, we'll render the markdown content with syntax highlighting
```
I thought I wanted to output markdown in user message as well. What do you think?

MH4GF reviewed (commented):
I wrote a comment!

MH4GF commented:
@kumanoayumi I would like to discuss this with you.

I do have a few concerns about the current “left = bot / right = user” bubble layout:

1. **Readability for code-heavy content**

   * Most replies from the AI will be SQL, DDL, or table-formatted Markdown. In the split-bubble layout those blocks get wrapped or require horizontal scrolling, which makes quick scanning and copy-pasting harder.
   * In the production build the chat will live in a resizable side-pane rather than a modal, so users can widen it, but even then a two-column bubble pattern still leaves \~30–40 % of the pane unused.

2. **“Conversation feel” vs. “deliverable focus”**

   * The alternating sides definitely convey a chat vibe, yet the primary user goal is to **obtain usable artefacts** (schemas, migrations, design notes). A document-style, single-column feed might serve that goal better by maximising horizontal space and reducing eye-zig-zag.

Would you be open to exploring a single-column timeline (speaker icon + colour for attribution) or a hybrid where code blocks automatically expand to full pane width? 

Let me know what you think!

kumanoayumi commented:
@MH4GF 

I think the hybrid approach—keeping user messages in the traditional bubble style while displaying only agent responses in a single-column format—makes a lot of sense.

Improved readability for deliverables like code and tables is a key benefit, and it maintains the conversational feel on the user side, which strikes a good balance.

We'll consider this further, taking into account visual flow and overall UI consistency!

kumanoayumi commented:
@MH4GF 

I've tried adjusting the design to use a single-column layout.
Along with the layout changes, I’ve updated the placement of avatars and timestamps to make each message unit clearer, and added labels for the user and agent names.

What do you think?

storybook: https://liam-storybook-git-ui-chatagent-message-route-06-core.vercel.app/?path=/docs/components-chat-agentmessage--docs&globals=backgrounds.value:!hex(333333)

https://github.com/user-attachments/assets/ff526d85-b323-4481-81c0-aa03cf02d492

app: https://liam-app-git-ui-chatagent-message-route-06-core.vercel.app/app/projects/c4341447-3192-45cd-bec6-79d5377b1600/ref/main/build

https://github.com/user-attachments/assets/41d8a0b5-190f-47a4-9c36-f9dc470be234

MH4GF commented:
@kumanoayumi Nice!! I think the message is now easy to read even with codes and tables. LGTM 👍🏻

MH4GF reviewed (approved):
LGTM 👍🏻
